pkg/api/v20230401/openshiftcluster_validatestatic.go

```diff
@@ -346,6 +367,45 @@ func (sv openShiftClusterStaticValidator) validateIngressProfile(path string, p
 	return nil
 }
 
+func (sv openShiftClusterStaticValidator) validateResourceTags(path string, t Tags) error {
+	if len(t) > maxTags {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path, fmt.Sprintf("The provided set of cluster resource tags is too large; it can contain at most %d tags.", maxTags))
+	}
+
+	var invalidTags []string
+
+	errorMsgComponents := []string{
+		"The following cluster resource tags either have an invalid name or an invalid value:\n",
+	}
+
+	for k, v := range t {
+		if !sv.resourceTagIsValid(k, v) {
+			invalidTags = append(invalidTags, k)
+			errorMsgComponents = append(errorMsgComponents, fmt.Sprintf("- %s\n", k))
+		}
+	}
+
+	errorMsgComponents = append(errorMsgComponents, `
+Tag names must:
+- Start with a letter
+- End with a letter, number, or underscore
+- Contain only letters, numbers, underscores, periods, and hyphens
+- Have length <= 128
+
+Tag values have no character restrictions, but must have length <= 256.`)
+
+	if len(invalidTags) > 0 {
+		errorMsg := strings.Join(errorMsgComponents, "")
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path, errorMsg)
+	}
+
```

Comment:
how about something like this ? 

```suggestion
	for k, v := range t {
		if !sv.resourceTagIsValid(k, v) {
			invalidTags = append(invalidTags, k)
			errorMsgComponents = append(errorMsgComponents, fmt.Sprintf("- %s\n", k))
		}
	}

	if len(invalidTags) > 0 {
		errorTemplate := `The following cluster resource tags either have an invalid name or an invalid value:
	- %s
Tag names must:
	- Start with a letter
	- End with a letter, number, or underscore
	- Contain only letters, numbers, underscores, periods, and hyphens
	- Have length <= 128
Tag values have no character restrictions, but must have length <= 256.
	`
		errorMsg := fmt.Sprintf(errorTemplate, strings.Join(invalidTags, "\n\t- "))
		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path, errorMsg)
	}

```

Reply:
That makes sense. The tabs help with readability too. I've changed it to be like this.